File: easy/3136_valid_word/solution.cpp
using namespace std;

#include <string>

class Solution {
public:
  bool isVowel(char c) {
    return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
  }

  bool isValid(string word) {
    if (word.size() < 3)
      return false;

    bool containsVowel = false;
    bool containsConsonant = false;

    for (char &c : word) {
      if (!isalnum(c))
        return false;

      c = tolower(c);

      if (!containsVowel && isVowel(c))
        containsVowel = true;
      if (!isVowel(c) && !isdigit(c))
        containsConsonant = true;
    }

    return containsVowel && containsConsonant;
  }
};

File: easy/3136_valid_word/solution.py
class Solution:
    def isValid(self, word: str) -> bool:
        if len(word) < 3:
            return False

        if not word.isalnum():
            return False

        vowels_set = set("aeiouAEIOU")

        contains_vowel = False
        contains_consonant = False

        for c in word:
            if not contains_vowel and c in vowels_set:
                contains_vowel = True
            if c not in vowels_set and not c.isdigit():
                contains_consonant = True

            if contains_vowel and contains_consonant:
                return True

        return False
